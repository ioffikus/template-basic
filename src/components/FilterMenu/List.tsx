import React from 'react';
import { Radio } from 'antd';
import _ from 'lodash';

interface IProps {
  ui: Alicanto.Models.UI;
  value: number[];
  fieldSlug: string;
  filterType: string;
  valueType: string;
  categoriesResults: Alicanto.Models.Category[];
  onChange: (value: number[], fieldSlug: string, valueType: string, filterType: string) => void;
}

interface IState {
  value: number;
}

class List extends React.PureComponent<IProps, IState> {
  readonly defauldValue = 'all';

  handleChange = (event: React.SyntheticEvent<any>) => {
    const value: 'all' | number = _.get(event.target, 'value', 'all');
    const nValue = value !== 'all' ? [value] : null;
    this.props.onChange(nValue, this.props.fieldSlug, this.props.valueType, this.props.filterType);
  };

  render() {
    const commonUI = this.props.ui.common;
    let value: string | number = this.defauldValue;
    if (!_.isNil(this.props.value)) {
      value = this.props.value[0];
    }

    return (
      <div className="root">
        <style jsx>{`
          .root :global(.filters) {
            max-width: 100%;
          }
          .root :global(.filters .ant-radio-button-wrapper) {
            display: flex;
            align-items: center;
            line-height: 1.5;
            height: auto;
            min-height: 20px;
            white-space: normal;
            padding-top: 4px;
            padding-bottom: 4px;
          }
          .root :global(.filters .ant-radio-button-wrapper + .ant-radio-button-wrapper) {
            margin-top: 10px;
          }
        `}</style>
        <Radio.Group onChange={this.handleChange} value={value} className="filters">
          <Radio.Button value={this.defauldValue}>
            {commonUI.filters.categories.items.allCategories.i18n.defaultText}
          </Radio.Button>
          {this.props.categoriesResults.map((item: Alicanto.Models.Category) => (
            <Radio.Button key={item.id} value={item.id}>
              {item.name}
            </Radio.Button>
          ))}
        </Radio.Group>
      </div>
    );
  }
}

export default List;
