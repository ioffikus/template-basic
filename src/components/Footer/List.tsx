import React from 'react';
import { Row, Col } from 'antd';
import Item from './Item';

interface IProps {
  ui: Alicanto.Models.UI;
}

interface IState {}

class List extends React.Component<IProps, IState> {
  render() {
    const items = this.props.ui.common.footerLinks.items;
    return (
      <div className="root app-container">
        <style jsx>{`
          .root :global(.item) {
            flex: 1;
          }
        `}</style>
        <Row type="flex">
          {Object.keys(items).map(key => {
            const item = items[key];
            return (
              <Col className="item" key={key}>
                <Item links={item.links} title={item.title.i18n.defaultText} />
              </Col>
            );
          })}
        </Row>
      </div>
    );
  }
}

export default List;
