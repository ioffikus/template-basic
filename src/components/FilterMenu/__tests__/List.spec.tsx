import React from 'react';
import { shallow } from 'enzyme';
import { Radio } from 'antd';
import uiStateMock from 'src/store/__mocks__/uiStore';
import filterStateMock from 'src/store/__mocks__/filterStore';
import categoriesStateMock from 'src/store/__mocks__/categoriesStore';
import List from '../List';

describe(null, () => {
  it('render() correct with initialState', () => {
    const listMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = listMock;
    const categoriesResults: Alicanto.Models.Category[] = [];
    const value: number[] = null;
    const onChange = jest.fn();
    const wrapper = shallow(
      <List
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        categoriesResults={categoriesResults}
        onChange={onChange}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with categories', () => {
    const listMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = listMock;
    const categoriesResults = categoriesStateMock.results;
    const value: number[] = null;
    const onChange = jest.fn();
    const wrapper = shallow(
      <List
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        categoriesResults={categoriesResults}
        onChange={onChange}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render() with selected value is different from default', () => {
    const listMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = listMock;
    const categoriesResults = categoriesStateMock.results;
    const value = [categoriesResults[0].id];
    const onChange = jest.fn();
    const wrapper = shallow(
      <List
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        categoriesResults={categoriesResults}
        onChange={onChange}
      />,
    );
    const radioGroupInstance = wrapper
      .find(Radio.Group)
      .dive()
      .instance() as Radio;
    expect(radioGroupInstance.props.value).toEqual(value[0]);
    expect(wrapper).toMatchSnapshot();
  });
  it('List.onChange()', () => {
    const listMock = filterStateMock.results[2];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = listMock;
    const categoriesResults = categoriesStateMock.results;
    const value = [categoriesResults[0].id];
    const newValue = categoriesResults[1].id;
    const onChange = jest.fn();
    const wrapper = shallow(
      <List
        ui={uiStateMock}
        value={value}
        fieldSlug={fieldSlug}
        valueType={valueType}
        filterType={filterType}
        categoriesResults={categoriesResults}
        onChange={onChange}
      />,
    );
    const radioGroupInstance = wrapper
      .find(Radio.Group)
      .dive()
      .instance() as Radio;
    radioGroupInstance.props.onChange({ target: { value: newValue } } as any);
    expect(onChange).toBeCalledWith([newValue], fieldSlug, valueType, filterType);
  });
});
