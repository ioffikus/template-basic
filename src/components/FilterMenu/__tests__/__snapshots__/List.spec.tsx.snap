// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct with categories 1`] = `
<div
  className="jsx-93181092 root"
>
  <JSXStyle
    css=".root.jsx-93181092 .filters{max-width:100%;}.root.jsx-93181092 .filters .ant-radio-button-wrapper{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;line-height:1.5;height:auto;min-height:20px;white-space:normal;padding-top:4px;padding-bottom:4px;}.root.jsx-93181092 .filters .ant-radio-button-wrapper + .ant-radio-button-wrapper{margin-top:10px;}"
    styleId="93181092"
  />
  <RadioGroup
    className="filters"
    disabled={false}
    onChange={[Function]}
    value="all"
  >
    <RadioButton
      prefixCls="ant-radio-button"
      value="all"
    >
      All categories
    </RadioButton>
    <RadioButton
      key="23"
      prefixCls="ant-radio-button"
      value={23}
    >
      Adventure novel
    </RadioButton>
    <RadioButton
      key="29"
      prefixCls="ant-radio-button"
      value={29}
    >
      Biography
    </RadioButton>
    <RadioButton
      key="35"
      prefixCls="ant-radio-button"
      value={35}
    >
      books
    </RadioButton>
    <RadioButton
      key="37"
      prefixCls="ant-radio-button"
      value={37}
    >
      eBooks
    </RadioButton>
  </RadioGroup>
</div>
`;

exports[`render() correct with initialState 1`] = `
<div
  className="jsx-93181092 root"
>
  <JSXStyle
    css=".root.jsx-93181092 .filters{max-width:100%;}.root.jsx-93181092 .filters .ant-radio-button-wrapper{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;line-height:1.5;height:auto;min-height:20px;white-space:normal;padding-top:4px;padding-bottom:4px;}.root.jsx-93181092 .filters .ant-radio-button-wrapper + .ant-radio-button-wrapper{margin-top:10px;}"
    styleId="93181092"
  />
  <RadioGroup
    className="filters"
    disabled={false}
    onChange={[Function]}
    value="all"
  >
    <RadioButton
      prefixCls="ant-radio-button"
      value="all"
    >
      All categories
    </RadioButton>
  </RadioGroup>
</div>
`;

exports[`render() with selected value is different from default 1`] = `
<div
  className="jsx-93181092 root"
>
  <JSXStyle
    css=".root.jsx-93181092 .filters{max-width:100%;}.root.jsx-93181092 .filters .ant-radio-button-wrapper{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;line-height:1.5;height:auto;min-height:20px;white-space:normal;padding-top:4px;padding-bottom:4px;}.root.jsx-93181092 .filters .ant-radio-button-wrapper + .ant-radio-button-wrapper{margin-top:10px;}"
    styleId="93181092"
  />
  <RadioGroup
    className="filters"
    disabled={false}
    onChange={[Function]}
    value={23}
  >
    <RadioButton
      prefixCls="ant-radio-button"
      value="all"
    >
      All categories
    </RadioButton>
    <RadioButton
      key="23"
      prefixCls="ant-radio-button"
      value={23}
    >
      Adventure novel
    </RadioButton>
    <RadioButton
      key="29"
      prefixCls="ant-radio-button"
      value={29}
    >
      Biography
    </RadioButton>
    <RadioButton
      key="35"
      prefixCls="ant-radio-button"
      value={35}
    >
      books
    </RadioButton>
    <RadioButton
      key="37"
      prefixCls="ant-radio-button"
      value={37}
    >
      eBooks
    </RadioButton>
  </RadioGroup>
</div>
`;
